src/components/CountryPicker/index.js

```diff
@@ -0,0 +1,92 @@
+import React, {useState, useEffect} from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import styles from '../../styles/styles';
+import MenuItemWithTopDescription from '../MenuItemWithTopDescription';
+import * as PersonalDetails from '../../libs/actions/PersonalDetails';
+import useLocalize from '../../hooks/useLocalize';
+import CountrySelectorModal from './CountrySelectorModal';
+import FormHelpMessage from '../FormHelpMessage';
+
+const propTypes = {
+    /** The ISO code of the country */
+    countryISO: PropTypes.string,
+
+    /** Form Error description */
+    errorText: PropTypes.string,
+
+    /** Country to display */
+    // eslint-disable-next-line react/require-default-props
+    value: PropTypes.string,
+
+    /** Callback to call when the input changes */
+    onInputChange: PropTypes.func,
+
+    /** Callback to call when the Country is updated */
+    onCountryUpdated: PropTypes.func,
+};
+
+const defaultProps = {
+    countryISO: '',
+    errorText: '',
+    onInputChange: () => {},
+    onCountryUpdated: () => {},
+};
+
+const CountryPicker = React.forwardRef(({value, countryISO, errorText, onCountryUpdated, onInputChange}, ref) => {
+    const {translate} = useLocalize();
+    const [isPickerVisible, setIsPickerVisible] = useState(false);
+    const countryValue = value || countryISO || '';
+
+    const showPickerModal = () => {
+        setIsPickerVisible(true);
+    };
+
+    const hidePickerModal = () => {
+        setIsPickerVisible(false);
+    };
+
+    const onStateSelected = (state) => {
+        onInputChange(state.value);
+        hidePickerModal();
+    };
+
+    useEffect(() => {
+        if (!onCountryUpdated || countryValue.length === 0) {
+            return;
+        }
+        onCountryUpdated(countryValue);
+    }, [countryValue, onCountryUpdated]);
+
+    const title = PersonalDetails.getCountryName(countryValue);
+    const descStyle = title.length === 0 ? styles.addressPickerDescription : null;
+
+    return (
+        <View>
+            <MenuItemWithTopDescription
+                ref={ref}
+                shouldShowRightIcon
+                title={title}
+                descriptionTextStyle={descStyle}
+                description={translate('common.country')}
+                onPress={showPickerModal}
+            />
+            <View style={styles.ml5}>
+                <FormHelpMessage message={errorText} />
+            </View>
+            <CountrySelectorModal
+                key={countryValue}
```

Comment:
I need to reset the state of `CountrySelectorModal`

Reply:
Sorry but why we need to reset the state? Changing the key will render a new component (not a good thing performance speaking). What would go wrong if we omit the key?

Reply:
Because the modal is now part of the whole component, the `useState` keeps the initial value, so if the user types "d" and choose `Delaware`, the next time that the modal opens will show "d". 
Did some profiling it wasn't that bad, but to avoid making the component recreated from scratch I lift the state up

Reply:
Instead of moving the state and to keep things consist let's do it the same way it's done in `YearPickerModal`. There is a `useEffect` that clears the input after hiding the modal.

```js
    useEffect(() => {
        if (props.isVisible) {
            return;
        }
        setSearchText('');
    }, [props.isVisible]);
```

Reply:
But I don't need to clear the value, I need to get the selected value, which could be `KS`, and set it as state "Kansas" so that can be displayed.
Also, because we are planning to move away from `lodashGet` I prefer to keep it in the same file instead of keep spreading it, so I prefer this way than adding a function in the middle

Reply:
Makes sense :+1: